polaris-core/src/main/java/org/apache/polaris/core/storage/StorageUtil.java

```diff
@@ -62,4 +68,50 @@ public class StorageUtil {
   public static @Nonnull String getBucket(URI uri) {
     return uri.getAuthority();
   }
+
+  /** Given a TableMetadata, extracts the locations where the table's [meta]data might be found. */
+  public static @Nonnull Set<String> getLocationsAllowedToBeAccessed(TableMetadata tableMetadata) {
+    return getLocationsAllowedToBeAccessed(tableMetadata.location(), tableMetadata.properties());
+  }
+
+  /** Given a baseLocation and entity (table?) properties, extracts the relevant locations */
+  public static @Nonnull Set<String> getLocationsAllowedToBeAccessed(
+      String baseLocation, Map<String, String> properties) {
+    Set<String> locations = new HashSet<>();
+    locations.add(baseLocation);
+    if (properties.containsKey(IcebergTableLikeEntity.USER_SPECIFIED_WRITE_DATA_LOCATION_KEY)) {
+      locations.add(properties.get(IcebergTableLikeEntity.USER_SPECIFIED_WRITE_DATA_LOCATION_KEY));
+    }
+    if (properties.containsKey(IcebergTableLikeEntity.USER_SPECIFIED_WRITE_METADATA_LOCATION_KEY)) {
+      locations.add(
+          properties.get(IcebergTableLikeEntity.USER_SPECIFIED_WRITE_METADATA_LOCATION_KEY));
+    }
+    return removeRedundantLocations(locations);
+  }
+
+  /** Given a ViewMetadata, extracts the locations where the view's [meta]data might be found. */
+  public static @Nonnull Set<String> getLocationsAllowedToBeAccessed(ViewMetadata viewMetadata) {
+    return Set.of(viewMetadata.location());
+  }
+
+  /** Removes "redundant" locations, so {/a/b/, /a/b/c, /a/b/d} will be reduced to just {/a/b/} */
+  private static @Nonnull Set<String> removeRedundantLocations(Set<String> locationStrings) {
+    HashSet<String> result = new HashSet<>(locationStrings);
```

Comment:
Since this is a new collection, it can be produced by
```java
locationStrings.stream()
  .filter(Objects::nonNull)
  .map(StorageLocation::of)
  .collect(Collectors.toCollection(HashSet::new));
```

Reply:
Yes, but you'd save the exponential instantiation of SotrageLocation objects

Reply:
Actually you could safe the inner loop with a sorted collection.

Reply:
Oh true, let's try to save the extra `StorageLocation` calls. Just pushed an update, let me know what you think!

I don't think we need to get too crazy trying to optimize this loop, considering the real number of locations is <=3 in practice and that we loop over them elsewhere when doing the overlap check.

Reply:
Looked a bit deeper into this and figured out some things need to be clarified in the Javadocs.

Although this reduction is technically not needed for Azure (actually: not possible :sigh:), it's probably worth to have some tests for this.

Opened https://github.com/eric-maynard/polaris/pull/5 with what I had in mind, feel free to take what you want/need.

Reply:
The main change to the javadocs seems to be comments like `Must be overridden by subclasses`, but I don't think that's true. In fact it's the opposite -- to date we've only created subclasses where we wish to override, but we could create subclasses where it's not necessary. For example, a `LocalStorageLocation`.

Reply:
True, "must" might be too strong. The implementations may however be very specific to the storage-type - that's what I've been thinking of.